lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -272,17 +272,23 @@ void DeclAndTypeClangFunctionPrinter::printFunctionSignature(
       os << ", ";
     HasParams = true;
     interleaveComma(additionalParams, os, [&](const AdditionalParam &param) {
-      assert(param.role == AdditionalParam::Role::Self);
-      CFunctionSignatureTypePrinterModifierDelegate delegate;
-      delegate.prefixIndirectParamValueTypeInC = [](raw_ostream &os) {
-        os << "SWIFT_CONTEXT ";
-      };
-      if (param.isIndirect) {
-        (*delegate.prefixIndirectParamValueTypeInC)(os);
-        os << "void * _Nonnull _self";
-      } else {
-        print(param.type, OptionalTypeKind::OTK_None, "_self",
-              /*isInOut*/ false, delegate);
+      if (param.role == AdditionalParam::Role::Self) {
+        CFunctionSignatureTypePrinterModifierDelegate delegate;
+        delegate.prefixIndirectParamValueTypeInC = [](raw_ostream &os) {
+          os << "SWIFT_CONTEXT ";
+        };
+        if (FD->hasThrows())
+          os << "SWIFT_CONTEXT ";
```

Comment:
Do you need this here? We already add SWIFT_CONTEXT.

Reply:
At least on the swift-functions-errors* tests, the SWIFT_CONTEXT that we already have was never printed...